### PR TITLE
Implement the new schema

### PR DESCRIPTION
**Pull request by Andrew-Dickinson**
Migrates to the new schema design. Primary changes are the addition of `Node` and `Device` tables

This forced an overhaul of nearly everything. That overhaul is incomplete, here's an (incomplete) task summary:
- [x] Schema changes
- [x] Get tests passing
- [ ] Admin UI overhaul
- [ ] Import script overhaul (see #224 )
- [ ] Fix merge conflicts with #216 
- [ ] Address-based de-duplication in join form
- [ ] KML & Map correctness checks
- [ ] Query & Lookup forms need some love (and tests)

### CODE REVIEW

File: src/meshapi/models/devices/device.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    )
+
+    name = models.CharField(
+        default=None,
+        blank=True,
+        null=True,
+        help_text="The colloquial name of this node used among mesh volunteers, if applicable",
+    )
+
+    model_name = models.CharField(
```
Comment (WillNilges) [inline src/meshapi/models/devices/device.py:44]:
```suggestion
    model = models.CharField(
```

File: src/meshapi/models/devices/sector.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            MaxValueValidator(360),
+        ],
+    )
+
+    ssid = models.CharField(
+        default=None,
+        blank=True,
+        null=True,
+        help_text="The SSID being broadcast by this device",
+    )
```
Comment (WillNilges) [inline src/meshapi/models/devices/sector.py:32]: Consider moving this to device

File: src/meshapi/models/link.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    class LinkStatus(models.TextChoices):
+        INACTIVE = "Inactive"
+        PLANNED = "Planned"
+        ACTIVE = "Active"
+
+    class LinkType(models.TextChoices):
+        STANDARD = "Standard"
+        VPN = "VPN"
+        MMWAVE = "MMWave"
+        FIBER = "Fiber"
```
Comment (WillNilges) [inline src/meshapi/models/link.py:16]: Should we preemptively add some more link types?
- Wired
- 24GHz
- 5GHz
- Etc?

File: src/meshapi/views/forms.py [hunk 4]
```diff
@@ -185,6 +185,7 @@ def join_form(request):
         member=join_form_member,
         referral=r.referral,
         notes=None,
+        node=join_form_building.primary_node if join_form_building.primary_node else None,
```
Comment (WillNilges) [inline src/meshapi/views/forms.py:188]: ok

File: src/meshapi/views/forms.py [hunk 5]
```diff
@@ -386,19 +388,30 @@ def network_number_assignment(request):
...
+        node_object = Node(
+            network_number=free_nn,
+            status=Node.NodeStatus.ACTIVE,
+            latitude=nn_building.latitude,
+            longitude=nn_building.longitude,
+            altitude=nn_building.altitude,
+            install_date=date.today(),
+            notes="Created by NN Assignment form",
+        )
+        node_object.save()
```
Comment (WillNilges) [inline src/meshapi/views/forms.py:408]: You should try/catch saving the node object

**Review by WillNilges — APPROVED** [review]
ok

File: src/meshapi/models/link.py (hunk 3 above)
Comment (Andrew-Dickinson) [reply src/meshapi/models/link.py:16]: Not at this time. Easy to play with in the future

**Pull request merged**